ibis/backends/pandas/execution/window.py

```diff
@@ -338,6 +341,8 @@ def execute_window_op(
     )
     series = post_process(result, data, ordering_keys, grouping_keys)
 
+    if timecontext:
+        series = construct_multi_index_series(series, data)
```

Comment:
Are the series and data matching in row ordering here?

Reply:
i.e. I think the series is ordered in group_by ordering, and the data is ordered in original ordering (probably ordered by time).

Reply:
It's a good point. I made some modifications here. Time index should be injected to series prior to groupby. I add logics in `aggcontext.py` for this, row order is well tracked in `agg` phase.